packages/react-combobox/src/components/Combobox/useCombobox.ts

```diff
@@ -11,18 +23,167 @@ import type { ComboboxProps, ComboboxState } from './Combobox.types';
  * @param props - props from this instance of Combobox
  * @param ref - reference to root HTMLElement of Combobox
  */
-export const useCombobox_unstable = (props: ComboboxProps, ref: React.Ref<HTMLElement>): ComboboxState => {
+export const useCombobox_unstable = (
+  props: ComboboxProps,
+  optionCollection: OptionCollectionState,
+  ref: React.Ref<HTMLButtonElement>,
+): ComboboxState => {
+  const {
+    inline = false,
+    multiselect,
+    onOpenChange,
+    open: controlledOpen,
+    placeholder,
+    positioning,
+    value: controlledValue,
+  } = props;
+  const {
+    options,
+    collectionData: { count, getOptionAtIndex, getIndexOfKey, getOptionByKey },
+  } = optionCollection;
+  const idBase = useId('combobox');
+
+  const [activeOption, setActiveOption] = React.useState<OptionValue | undefined>();
+  const [selectedKeys, selectKey] = useSelection(props);
+
+  const [value, setValue] = useControllableState({
+    state: controlledValue,
+    initialState: undefined,
+  });
+
+  const [open, setOpenState] = useControllableState({
+    state: controlledOpen,
+    initialState: false,
+  });
+
+  // popper
+  const popperOptions = {
+    enabled: open,
```

Comment:
is neabled needed here ? The only difference is that the listbox is aways rendered in DOM, which is not essential for for combobox a11y AFAIK.

Also in the the `renderCombobox` function, you use `open` in a ternary to render the listbox... which actually makes `enabled` useless here.